src/test/java/com/google/api/generator/engine/ast/CastExprTest.java

```diff
@@ -123,4 +169,13 @@ public void invalidCastExpr_castNumericToBoolean() {
         IllegalStateException.class,
         () -> CastExpr.builder().setType(TypeNode.BOOLEAN).setExpr(variableExpr).build());
   }
+
+  @Test
+  public void invalidCastExpr_castToVoidType() {
+    Variable variable = Variable.builder().setName("x").setType(TypeNode.VOID_OBJECT).build();
```

Comment:
This looks like it tests `(void) x` where `x` is a void variable. Casting an already-void-typed variable will fail, but what about casting an int-typed variable?

Reply:
Updated the unit test to `int x =(void) x` . This fails for the first condition check, which does not allow `void` in castType or exprType.